### PR TITLE
Disables finalize after segfaulting error

### PR DESCRIPTION
**Pull request by morris25**
Also adds `LZ4Exception`.
In response to #14

### CODE REVIEW

File: src/stream_compression.jl [hunk 1]
```diff
@@ -38,7 +28,7 @@ Creates an LZ4 compression codec.
 - `autoflush::Bool=false`: always flush if `true`
 """
 function LZ4Compressor(; kwargs...)
-    x, y = splitkwargs(kwargs, (:compressionlevel, :autoflush))
+    x, y = TranscodingStreams.splitkwargs(kwargs, (:compressionlevel, :autoflush))
```
Comment (omus) [inline src/stream_compression.jl]: Maybe just do `using TranscodingStreams: splitkwargs` at the top of the file?

File: src/stream_compression.jl [hunk 2]
```diff
@@ -138,8 +128,9 @@ function TranscodingStreams.process(codec::LZ4Compressor, input::Memory, output:
 
 end
 
-struct LZ4Decompressor <: TranscodingStreams.Codec
+mutable struct LZ4Decompressor <: TranscodingStreams.Codec
     dctx::Ref{Ptr{LZ4F_dctx}}
+    panic_mode::Bool
```
[the remark this answers is not in the document]
Comment (morris25) [reply src/stream_compression.jl]: That is a much better idea, yes.

File: src/stream_compression.jl [hunk 3]
```diff
@@ -138,7 +128,7 @@ function TranscodingStreams.process(codec::LZ4Compressor, input::Memory, output:
 
 end
 
-struct LZ4Decompressor <: TranscodingStreams.Codec
+mutable struct LZ4Decompressor <: TranscodingStreams.Codec
     dctx::Ref{Ptr{LZ4F_dctx}}
 end
```
Comment (omus) [inline src/stream_compression.jl]: We should switch this back to being non-mutable. You'll be able to modify the reference still.

File: src/stream_compression.jl [hunk 4]
```diff
@@ -198,6 +188,9 @@ function TranscodingStreams.process(codec::LZ4Decompressor, input::Memory, outpu
         end
 
     catch err
+        if typeof(err) == LZ4Exception && err.msg == "ERROR_frameType_unknown"
```
Comment (omus) [inline src/stream_compression.jl]:
```julia
if isa(err, LZ4Exception) && ...
```

**Pull request merged**